scripts/50_update_files.sh

```diff
@@ -32,10 +35,39 @@ usage()
     echo "adding them to the scripts runtime environment."
 }
 
+cleanup()
+{
+    # On slow media, umount and/or rmdir can fail with 'resource busy' errors.
+    # To do our best with cleanup, attempt this a few times before giving up.
+    failed=0
+    while test -d "${UPDATE_TARGET}"; do
+        if grep -q "${UPDATE_TARGET}" "/proc/mounts"; then
+            if ! umount "${UPDATE_TARGET}"; then
+                failed=$((failed + 1))
```

Comment:
quotes

Reply:
Could it happen that the script never returns?
In that case it could be safer to let a reboot clean up (because of temp) the mess if the umount and/or delete fails.

Reply:
it is possible for this to fail, if the device stays 'busy' for more then 15 to 30 seconds. The update script then exists with error 1.

so the question then is, do we want to increase the '-ge 30' to allow for more time on failure (which should recover eventually to be fair) (i increase it to 300 for max 10 minutes, in case the flash needs to write a lot) or what else?

The rsync + sync job should try to wait as long as possible, but if the MMC module crashes and just waits for ever. What is the safest step to continue with? Reboot probably not, as a reboot will leave the system in an unknown state. That's for start_update.sh to handle determine in case of a failed function ...

Reply:
The question is should the script fail if temporary stuff can not be cleaned up properly, because of limitations in the os?

Reply:
you missunderstand, this is not temporary stuff not being up properly, this is umount. And umount is failing because the device it is trying to umount is still busy.
Busy being very likely that its int he background still writing to the actual flash.

So reboot while the controller is still doing that, loose changes, have an undefined state on your filesystem, that you don't even know (corrupt files etc).

Reply:
Ah yes you are right, but shouldn't you only try to delete the mount point then when the actual umount succeeds?

Reply:
well since we are using rmdir, it will only work if it is not mounted :)